packages/blocks/src/store/process-block-type.js

```diff
@@ -102,13 +124,20 @@ export const processBlockType =
 			),
 		};
 
+		// Stabilize any experimental supports before applying filters.
+		blockType.supports = stabilizeSupports( blockType.supports );
+
 		const settings = applyFilters(
 			'blocks.registerBlockType',
 			blockType,
 			name,
 			null
 		);
 
+		// Re-stabilize any experimental supports after applying filters.
+		// This ensures that any supports updated by filters are also stabilized.
+		blockType.supports = stabilizeSupports( blockType.supports );
```

Comment:
This util is useful as it needs to run a few times. I see it only does any processing when they `typography` object is provided, so it should be fast enough.